amr_kitchen/plotfile_cooker.py

```diff
@@ -203,17 +203,55 @@ def __init__(self, fields, boxes, field_arg, limit_level):
             raise IndexError((f"The field indexing [{field_arg}] is not"
                               f" compatible with the number of fields"
                               f" in the plotfile ({len(fields)})"))
-        self.boxes = boxes
+        self.cells = cells
         self.fields = fields
         self.limit_level = limit_level
+        self.boxes = boxes
 
     def __getitem__(self, key):
         if key > self.limit_level:
             raise ValueError((f"The maximum AMR level of the plotfile"
                               f" is {self.limit_level}"))
-        return LevelDataStream(self.boxes[key]['files'],
-                               self.boxes[key]['offsets'],
+        return LevelDataStream(self.cells[key]['files'],
+                               self.cells[key]['offsets'],
                                self.farg)
+    
+    def __call__(self, cord_x, cord_y, cord_z = None):
+        if cord_z == None:
+            key = (cord_x, cord_y)
+        else:
+            key = (cord_x, cord_y, cord_z)
+
+        # Input validation
+        if self.boxes == None:
+            raise KeyError("PlotfileCooker object was not initiated with a True 'point' argument")
+        if len(key) != self.limit_level+1:
+            raise KeyError(f"Desired point {key} of {len(key)}d size doesn't correlate with the {self.limit_level+1}d size of the plotfile")
+
+        # Let's see in which box does the point belong 
+        matching_indicies = []
+        for box in range(len(self.boxes[self.limit_level])):
+            sous_box = self.boxes[self.limit_level][box]
+            for component in range(len(key)):
+                if key[component] < sous_box[component][0] or key[component] > sous_box[component][-1]:
+                    break
```

Comment:
You can use this synthax to find matches with fast numpy instead of loop:
```python
box_match = (boxes[:, 0, 0] >= point[0]) & \
                (point[0] <= boxes[:, 0, 1]) & \
                (boxes[:, 1, 0] >= point[1]) & \
                (point[1] <= boxes[:, 1, 1]) & \
                (boxes[:, 2, 0] >= point[2]) & \
                (point[2] <= boxes[:, 2, 1])
```